### PR TITLE
fix alignment

### PR DESCRIPTION
**Pull request by hana-alice**
Re: cocos-creator/3d-tasks#

Changelog:
 *

### CODE REVIEW

**Comment by hana-alice** [comment]
https://github.com/JamieBerghmans/Cylinder-iOS13/blob/master/deps/iPhoneOS13.2.sdk/usr/include/AppleTextureEncoder.h 
line436 mentioned 16-byte aligned, can't find any other docs request address start alignment on astc_rgba_4*4

**Comment by hana-alice** [comment]
@star-e @yiwenxue  align address to block size

File: native/cocos/renderer/gfx-agent/DeviceAgent.cpp [hunk 1]
```diff
@@ -284,23 +285,32 @@ void doBufferTextureCopy(const uint8_t *const *buffers, Texture *texture, const
     for (uint32_t i = 0U; i < count; i++) {
         bufferCount += regions[i].texSubres.layerCount;
     }
-    uint32_t totalSize = sizeof(BufferTextureCopy) * count + sizeof(uint8_t *) * bufferCount;
+    
+    auto alignTo = [](size_t size, uint32_t alignment) {
+        return (size + alignment - 1) / alignment * alignment;
+    };
+
```
Comment (star-e) [inline native/cocos/renderer/gfx-agent/DeviceAgent.cpp]: We might use boost::alignment::align_up
https://www.boost.org/doc/libs/1_79_0/boost/align/align_up.hpp

File: native/cocos/renderer/gfx-agent/DeviceAgent.cpp [hunk 2]
```diff
@@ -284,23 +285,32 @@ void doBufferTextureCopy(const uint8_t *const *buffers, Texture *texture, const
...
         bufferCount += regions[i].texSubres.layerCount;
     }
-    uint32_t totalSize = sizeof(BufferTextureCopy) * count + sizeof(uint8_t *) * bufferCount;
+    
+    auto alignTo = [](size_t size, uint32_t alignment) {
+        return (size + alignment - 1) / alignment * alignment;
+    };
+    
+    Format format = texture->getFormat();
+    uint32_t dataRequestAlignment = formatSize(format, 1, 1, 1); //block size
```
Comment (star-e) [inline native/cocos/renderer/gfx-agent/DeviceAgent.cpp]: It seems dataRequestAlignment is not necessary. Will removing dataRequestAlignment  cause trouble?

**Review by yiwenxue — APPROVED** [review]

**Review by star-e — APPROVED** [review]

**Pull request merged**